evm_loader/program/src/entrypoint.rs

```diff
@@ -565,6 +565,114 @@ fn process_instruction<'a>(
+                    operator_sol_info,
+                    collateral_pool_sol_info,
+                    system_info)?;
+                payment::transfer_from_operator_to_deposit(
+                    operator_sol_info,
+                    storage_info,
+                    system_info)?;
+
+                if trx.to.is_some() {
+                    do_partial_call(&mut storage, step_count, &account_storage, trx_accounts, trx.call_data, trx.value, trx_gas_limit)?;
+                } else {
+                    do_partial_create(&mut storage, step_count, &account_storage, trx_accounts, trx.call_data, trx.value, trx_gas_limit)?;
+                }
+
+                storage.block_accounts(program_id, trx_accounts)
+            }
+            else {
+                storage.check_accounts(program_id, trx_accounts)?;
+                let mut account_storage = ProgramAccountStorage::new(program_id, trx_accounts)?;
+                let call_return = do_continue(&mut storage, step_count, &mut account_storage, trx_accounts)?;
+
+                if let Some(call_results) = call_return {
+                    payment::transfer_from_deposit_to_operator(
+                        storage_info,
+                        operator_sol_info,
+                        system_info)?;
+                    if get_token_account_owner(operator_eth_info)? != *operator_sol_info.key {
+                        return Err!(ProgramError::InvalidInstructionData;
+                            "Wrong operator token ownership: operator token owner = {:?}, operator key = {:?}",
+                            operator_eth_info.owner,
+                            operator_sol_info.key)
+                    }
+                    let (gas_limit, gas_price) = storage.get_gas_params()?;
+                    let used_gas = call_results.1;
+                    if used_gas > gas_limit {
+                        return Err!(ProgramError::InvalidArgument);
+                    }
+                    let gas_price_wei = U256::from(gas_price);
+                    let fee = U256::from(used_gas)
+                        .checked_mul(gas_price_wei).ok_or_else(||E!(ProgramError::InvalidArgument))?;
+                    token::transfer_token(
+                        accounts,
+                        user_eth_info,
+                        operator_eth_info,
+                        account_storage.get_caller_account_info().ok_or_else(||E!(ProgramError::InvalidArgument))?,
+                        account_storage.get_caller_account().ok_or_else(||E!(ProgramError::InvalidArgument))?,
+                        &fee)?;
+
+                    applies_and_invokes(
+                        program_id,
+                        &mut account_storage,
+                        accounts,
+                        Some(operator_sol_info),
+                        call_results)?;
+
+                    storage.unblock_accounts_and_destroy(program_id, trx_accounts)?;
+                }
```

Comment:
Extract function from these lines. Exact same text in `Continue` and in `PartialCallOrContinueFromRawEthereumTX`

Reply:
Fixed